lib/model/model.go

```diff
@@ -840,10 +840,11 @@ func (m *model) Completion(device protocol.DeviceID, folder string) FolderComple
 // DBSnapshot returns a snapshot of the database content relevant to the given folder.
 func (m *model) DBSnapshot(folder string) (*db.Snapshot, error) {
 	m.fmut.RLock()
-	rf, ok := m.folderFiles[folder]
+	err := m.checkFolderRunningLocked(folder)
+	rf := m.folderFiles[folder]
 	m.fmut.RUnlock()
-	if !ok {
-		return nil, errFolderMissing
+	if err != nil {
+		return nil, err
```

Comment:
The desired outcome here is that we return `errFolderNotRunning` (from checkFolderRunning) instead of the current `errFolderMissing`?

Reply:
It feels like this thing might benefit from an actual API test

Reply:
Misunderstood your comment. The test I added is still useful in my opinion, but does nothing for the rest api. Which is indeed still returning a 404, and that needs a different workaround. I'll file a separate PR for that.